frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadFileFormContext.component.tsx

```diff
@@ -0,0 +1,47 @@
+/**
+ *  Copyright (C) 2023 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { createContext, useState } from 'react';
+
+export interface UploadsContextType {
+	selectedUploadId: string;
+	setSelectedUploadId: (id: string) => void;
+	originalIndex: number
+	setOriginalIndex: (index: number) => void;
+
+	props?: any;
+}
+
+const defaultValue: UploadsContextType = {
+	selectedUploadId: '',
+	setSelectedUploadId: () => {},
+	originalIndex: null,
+	setOriginalIndex: () => {},
+};
+export const UploadsContext = createContext(defaultValue);
+UploadsContext.displayName = 'UploadsContext';
+
+export const UploadsContextComponent = ({ children }) => {
+	const [selectedUploadId, setSelectedUploadId] = useState('');
```

Comment:
Yes. I am commenting here before either of you two can. I don't like this either, since whenever you update one state you update the other too, so surely you'd be able to combine them? But I couldn't figure out a clean way of doing that.

Basically it's a result of react-hook-forms using two different arrays when you use useFieldArray(). There is `field` which includes the `uploadId` and the values fetched by `getValues()` which includes all the up to date data. Some aspects of the form need the uploadId and some need the values. I thought maybe adding the uploadId to the values might be a good idea, but I don't know